doc/function.md

```diff
@@ -1,5 +1,255 @@
 # Function
 
-You are reading a draft of the next documentation and it's in continuous update so
-if you don't find what you need please refer to:
-[C++ wrapper classes for the ABI-stable C APIs for Node.js](https://nodejs.github.io/node-addon-api/)
+The **Function** class is a representation of the JavaScript function. A function
+is a JavaScript procedure, a set of statements that performs a task or calculates
+a value. This class provides some methods to create and exectue a function.
+
+The `Function` class inherits its behavior from the `Object` class (for more info
+see: [Object](object.md)).
+
+## Methods
+
+### Constructor
+
+Creates a new empty instance of `Function`.
+
+```cpp
+Function();
+```
+
+### Constructor
+
+Creates a new instance of `Function` object.
+
+```cpp
+Function(napi_env env, napi_value value);
+```
+
+- `[in] env`: The `napi_env` environment in which to construct the Value object.
+- `[in] value`: The C++ primitive from which to instantiate the Value. `value`
+may be any of:
+
+  - bool
+  - Any integer type
+  - Any floating point type
+  - const char* (encoded using UTF-8, null-terminated)
+  - const char16_t* (encoded using UTF-16-LE, null-terminated)
+  - std::string (encoded using UTF-8)
+  - std::u16string
+  - napi::Value
+  - napi_value
+
+Returns a non-empty `Function` instance.
+
+### New
+
+Creates instance of a `Function` object.
+
+```cpp
+template <typename Callable>
+static Function New(napi_env env, Callable cb, const char* utf8name = nullptr, void* data = nullptr);
+```
+
+- `[in] env`: The `napi_env` environment in which to construct the Function object.
+- `[in] cb`: Object that implement the operator **()** accepting a `const CallbackInfo&`
```

Comment:
Should this just say it needs to be an instance of Callable ?

Reply:
implement*s*

Reply:
is it good for you: *Object that implements Callable.*?